lib/Zonemaster/Backend/Validator.pm

```diff
@@ -6,7 +6,37 @@ use strict;
 use warnings;
 use 5.14.2;
 
+use Exporter qw( import );
 use JSON::Validator::Joi;
+use Readonly;
+
+our @EXPORT_OK = qw(
+  untaint_engine_type
+);
+
+our %EXPORT_TAGS = (
+    untaint => [
+        qw(
+          untaint_engine_type
+          )
+    ],
+);
+
+Readonly my $IPV4_RE => qr/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\$/;
+Readonly my $IPV6_RE => qr/^([0-9a-f]{1,4}:[0-9a-f:]{1,}(:[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})?)\$|([0-9a-f]{1,4}::[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})\$/i;
+
```

Comment:
Maybe a comment that these REs will accept more IPv4 and IPv6 addresses, respectively, than the legal ones and must be combined with some other method to check?